### ISSUE TITLE
Enum compression with the value tryFrom is always seems to be false

### ISSUE THREAD

**Issue by KVV0001** [kalessil/phpinspectionsea#1908]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate) |
| Language level | e.g. PHP 8.1 (check project settings or composer.json)                        |

Current behaviour
--
```php
RequestStatusEnum::COMPLETED === RequestStatusEnum::tryFrom($data['status'])'
``` 
[EA] 'RequestStatusEnum::COMPLETED === RequestStatusEnum::tryFrom($data['status'])' seems to be always false

Expected behaviour
--

Expect no errors

**Comment by ea-inspections-team** [comment]
Thank you for reporting @KVV0001!
The false positive has been resolved.

The root cause is PhpStorm not resolving return types properly for `tryFrom` and `from` enum methods.
We have implemented a workaround by adding custom type resolution for these methods.
